### ISSUE TITLE
a model property value is not synchronized from client to server

### ISSUE THREAD

**Issue by vlukashov** [vaadin/flow#3557]
As a developer I expect that model property values remain in sync between the client and the server. Currently (as of alpha 20 of Flow 1.0) this is not always the case.

The example below shows that the server-side value of the `checkboxChecked` model property does not get synced from the client at all - it remains `null` regardless of the changes done to this property on the client side.

```java
public interface ExampleModel extends TemplateModel {
    Boolean getCheckboxChecked();
}
```
```html
<template>
    <label>
        <input type="checkbox" checked="[[checkboxChecked]]" on-click="_onCheckboxClicked">
        a checkbox bound to a model property
    </label>
</template>
```
```javascript
class ExampleTemplate extends Polymer.Element {
    static get is() {
        return 'example-template'
    }

    static get properties() {
        return {
            checkboxChecked : {
                type: Boolean,
                observer: '_onCheckboxCheckedChanged'
            }
        };
    }

    _onCheckboxClicked(event) {
        this.checkboxChecked = event.target.checked;
    }
}
```

![screen recording 2018-02-14 at 03 56 pm](https://user-images.githubusercontent.com/22416150/36207841-dbf77e90-119f-11e8-886b-28a4ace247be.gif)

The source code of the demo project is attached: [platform18-model-properties-2.zip](https://github.com/vaadin/flow/files/1724225/platform18-model-properties-2.zip)

**Expected behavior**
 - client-side changes to the `checkboxChecked` property are synced to the server side (i.e. after checking the checkbox and clicking at the 'Check server-side value' button both lines below it should show `true`)

**Comment by Legioth** [comment]
What you are expecting is a fundamental security issue. Arbitrary JavaScript in the browser should not be allowed to modify server-side application state without opt-in from the developer.

In this case, you can opt-in by adding `@AllowClientUpdates` to `getCheckboxChecked()`. It would have worked automatically if there would have been a  two-way binding (i.e. `{{checkboxChecked}}`) in the template.

There was previously a notice about this in the server-side log, but further enhancements moved some of the logic to the client instead and the informative message was probably lost then. It might also be necessary to improve the documentation to make this situation more apparent.

**Comment by vlukashov** [comment]
Thanks for the explanation. For me this was not an expected behavior.

This ticket can be closed (apparently this is a feature and not a bug), but please create a ticket to improve the documentation and logging.

**Comment by denis-anisimov** [comment]
I think this ticket should be kept as is (or may be we already have it?): it's just about documentation but not the functionality.

**Comment by denis-anisimov** [comment]
Or this one can be considered as an original one: https://github.com/vaadin/flow/issues/3495

**Comment by pleku** [comment]
Closed in favor of #3495 for clarifying documentation
